compiler/quilt/tools/data_transfer.py

```diff
@@ -95,6 +95,33 @@ def _worker_thread():
                             try:
                                 existing_file_size = output_file.tell()
 
+                                # For zero-byte downloads, we don't need to resume, and range download
+                                # must be at least 1 anyways. 
+                                if total_bytes == 0:
```

Comment:
Actually, `total_bytes` is the total size of the whole package, not an individual file. That would be `original_size`.

(Also, technically, size might not be known cause some really old quilt clients don't set it... But it shouldn't happen in practice.)

Reply:
Hrm.. Thanks!

Reply:
Should we handle that case?  We could probably find a way to work around that.  Adds complexity, but if it's needed, it's needed.

Reply:
I'd punt on that. But, you should change the `or 0` code above to just raise an error.

Reply:
Are you familiar enough to know whether or not I can raise an error that recommends re-uploading the package?  I.e., would '--reupload' fix the issue?

Reply:
Familiar enough, and it's not worth worrying about. In prod, we have a total of 5 such files, created by us while testing stuff. And the clients that didn't have size support are so old that no one uses them anymore.

Reply:
Sweet.

Reply:
Now raises `QuiltException("Malformed package: No size for object " + obj_hash")`